openml-api/src/main/java/com/feedzai/openml/data/schema/DatasetSchema.java

```diff
@@ -135,6 +136,9 @@ public FieldSchema getTargetFieldSchema() {
      * @return the list of predictive fields.
      */
     public List<FieldSchema> getPredictiveFields() {
+        if (!getTargetIndex().isPresent()) {
```

Comment:
but this one I have to disagree:

here you are using `getTargetIndex()` but then you use (on the filter) the `this.targetIndex`. Is not obvious why you can do it for someone not familiar with the previous PR. can you reuse the result of the `getTargetIndex()` here (with a `.get` in case you prefer to keep the imperative style) ?

Reply:
Yup, I agree!